src/GraphQL/Resolvers/ObservableFromAsyncEnumerable.cs

```diff
@@ -0,0 +1,140 @@
+using GraphQLParser.AST;
+
+namespace GraphQL.Resolvers;
+
+/// <inheritdoc cref="ObservableFromAsyncEnumerable{T}.Create(Func{IResolveFieldContext, IAsyncEnumerable{T}})"/>
+internal sealed class ObservableFromAsyncEnumerable<T> : IObservable<object?>, IDisposable
+{
+    private readonly CancellationTokenSource _cancellationTokenSource;
+    private IAsyncEnumerable<T>? _enumerable;
+
+    private ObservableFromAsyncEnumerable(CancellationTokenSource cts, IAsyncEnumerable<T> enumerable)
+    {
+        _cancellationTokenSource = cts;
+        _enumerable = enumerable;
+    }
+
+    /// <summary>
+    /// Returns a source stream resolver delegate (which returns an <see cref="IObservable{T}"/>) from a delegate
+    /// which returns <see cref="IAsyncEnumerable{T}"/>. Each execution will create a new
+    /// <see cref="ObservableFromAsyncEnumerable{T}"/> instance which can only be used once.
+    /// </summary>
+    public static Func<IResolveFieldContext, ValueTask<IObservable<object?>>> Create(Func<IResolveFieldContext, ValueTask<IAsyncEnumerable<T>>> func)
+    {
+        return async context =>
+        {
+            var cts = CancellationTokenSource.CreateLinkedTokenSource(context.CancellationToken);
+            var enumerable = await func(new OverrideCancellationContext(context, cts.Token)).ConfigureAwait(false);
+            return new ObservableFromAsyncEnumerable<T>(cts, enumerable);
+        };
+    }
+
+    /// <inheritdoc cref="Create(Func{IResolveFieldContext, ValueTask{IAsyncEnumerable{T}}})"/>
+    public static Func<IResolveFieldContext, ValueTask<IObservable<object?>>> Create(Func<IResolveFieldContext, Task<IAsyncEnumerable<T>>> func)
+        => Create(context => new ValueTask<IAsyncEnumerable<T>>(func(context)));
+
+    /// <inheritdoc cref="Create(Func{IResolveFieldContext, ValueTask{IAsyncEnumerable{T}}})"/>
+    public static Func<IResolveFieldContext, ValueTask<IObservable<object?>>> Create(Func<IResolveFieldContext, IAsyncEnumerable<T>> func)
+        => Create(context => new ValueTask<IAsyncEnumerable<T>>(func(context)));
+
+    /// <inheritdoc/>
+    public IDisposable Subscribe(IObserver<object?> observer)
+    {
+        // note: this error should not occur, as GraphQL.NET does not call IObservable<T>.Subscribe
+        // more than once after executing the source stream resolver
+        var enumerable = Interlocked.Exchange(ref _enumerable, null)
+            ?? throw new InvalidOperationException("This method can only be called once.");
+
+        // this would only occur if IResolveFieldContext.CancellationToken has been signaled
+        _cancellationTokenSource.Token.ThrowIfCancellationRequested();
+
+        // iterate the async enumerable until the cancellation token is signaled via IDisposable.Dispose
+        SubscribeInternalAsync(observer, enumerable);
+
+        return this;
+    }
+
+    private async void SubscribeInternalAsync(IObserver<object?> observer, IAsyncEnumerable<T> enumerable)
```

Comment:
`async void` - not a frequent guest...

Reply:
The usage was directly copied from a MS implementation within System.Reactive, so I kept it, although almost all the rest of code changed.

Reply:
System.Reactive.Linq, that is.  They have code to convert an IAsyncEnumerable to IObservable which is what i started with.